pkg/token/service.go

```diff
@@ -124,10 +124,13 @@ func New(ctx context.Context, idCfg *config.IdentityConfig) (daemon.Daemon, erro
 	// If it is in refresh mode, when requesting tokens using the REST API for the domains and roles specified in TARGET_DOMAIN_ROLES,
 	// the cache is updated to ensure a cache hit from the first request.
 	if !idCfg.Init || enableWriteFiles {
-		// TODO: if cap(errs) == len(errs), implies all token updates failed, should be fatal
-		for _, err := range ts.updateTokenCachesAndWriteFiles(ctx, config.DEFAULT_MAX_ELAPSED_TIME_ON_INIT) {
+		errs := ts.updateTokenCachesAndWriteFiles(ctx, config.DEFAULT_MAX_ELAPSED_TIME_ON_INIT)
+		for _, err := range errs {
 			log.Errorf("Failed to refresh tokens after multiple retries: %s", err.Error())
 		}
+		if len(errs) != 0 && idCfg.Init {
+			return nil, fmt.Errorf("Unable to write token files for init mode")
```

Comment:
Since this terminates process\,  we need a fatal log error with the following:
- What happened
- Why happened
- What should end users do?

Reply:
fix: [10a7dc3](https://github.com/AthenZ/k8s-athenz-sia/pull/156/commits/10a7dc3a2f901e244d5d93d8ce51cf348146d216)